internal/gke/cluster_info.go

```diff
@@ -223,6 +224,42 @@ func fillClusterInfo(ctx context.Context, cluster, region string, cc CloudConfig
 	}, nil
 }
 
+// GetClusterInfos returns information about all GKE clusters running in a
+// given cloud project.
+// REQUIRES: The caller is running on the user machine.
+func GetClusterInfos(ctx context.Context, config CloudConfig) ([]*ClusterInfo, error) {
+	// Run `gcloud clusters list --format "value(name, location)"` to get the
+	// name and location of every cluster in the project.
+	out, err := runGcloud(config, "", cmdOptions{},
+		"container", "clusters", "list", "--format", "value(name, location)")
```

Comment:
It would be better to output in JSON and then decode into a struct, rather than parse strings a lines. WDYT?

Welcome to the world of gcloud hackery :)

Reply:
Great idea. Done!